### PR TITLE
[Feat] Affine scale with non-isotropic values

### PR DESCRIPTION
**Pull request by shijianjian**
### Description
Enable non-isotropic values. #615 

Usage:
```python
kornia.augmentation.RandomAffine(360, scale=(0., 1., 0., 1.))
kornia.augmentation.RandomAffine3D(360, scale=((0., 1.), (0., 1.), (0., 1.)))
```

### Status
**Ready/Work in progress/Hold**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or new feature that would cause existing functionality to change)
- [ ] New tests added to cover the changes
- [ ] Docstrings/Documentation updated

### PR Implementer
This is a small checklist for the implementation details of this PR.

If there are any questions regarding code style or other conventions check out our 
[summary](https://github.com/kornia/kornia/blob/master/CONTRIBUTING.rst).

- [ ] Did you discuss the functionality or any breaking changes before ?
- [ ] **Pass all tests**: did you test in local ? `make test`
- [ ] Unittests: did you add tests for your new functionality ?
- [ ] Documentations: did you build documentation ? `make build-docs`
- [ ] Implementation: is your code well commented and follow conventions ? `make lint`
- [ ] Docstrings & Typing: has your code documentation and typing ? `make mypy`
- [ ] Update notebooks & documentation if necessary

### KorniaTeam
<details>
  <summary>KorniaTeam workflow</summary>
  
  - [ ] Assign correct label
  - [ ] Assign PR to a reviewer
  - [ ] Does this PR close an Issue? (add `closes #IssueNumber` at the bottom if 
        not already in description)

</details>

### Reviewer
<details>
  <summary>Reviewer workflow</summary>

  - [ ] Do all tests pass? (Unittests, Typing, Linting, Documentation, Environment)
  - [ ] Does the implementation follow `kornia` design conventions?
  - [ ] Is the documentation complete enough ?
  - [ ] Are the tests covering simple and corner cases ?
 
</details>

### CODE REVIEW

File: kornia/geometry/transform/projwarp.py [hunk 1]
```diff
@@ -110,7 +110,10 @@ def get_projective_transform(center: torch.Tensor, angles: torch.Tensor, scales:
 
     # create rotation matrix
     angle_axis_rad: torch.Tensor = K.deg2rad(angles)
-    rmat: torch.Tensor = K.angle_axis_to_rotation_matrix(angle_axis_rad) * scales.view(-1, 1, 1)  # Bx3x3
+    rmat: torch.Tensor = K.angle_axis_to_rotation_matrix(angle_axis_rad)  # Bx3x3
+    scaling_matrix: torch.Tensor = torch.zeros((3, 3)).fill_diagonal_(1).repeat(rmat.size(0), 1, 1)
```
Comment (edgarriba) [inline kornia/geometry/transform/projwarp.py:114]: why not use `torch.eye` or [`kornia.eye_like`](https://kornia.readthedocs.io/en/latest/geometry.epipolar.html?highlight=eye_like#kornia.geometry.epipolar.eye_like) instead ?

File: kornia/geometry/transform/projwarp.py [hunk 2]
```diff
@@ -110,7 +110,10 @@ def get_projective_transform(center: torch.Tensor, angles: torch.Tensor, scales:
...
+    scaling_matrix = scaling_matrix * scales.unsqueeze(dim=2).repeat(1, 1, 3)
```
Comment (edgarriba) [inline kornia/geometry/transform/projwarp.py:115]: broadcasting not working here ? if scales come as shape Bx3 , you just need to .unsqueeze(1)

File: kornia/utils/grid.py [hunk 3]
```diff
@@ -28,12 +28,13 @@ def create_meshgrid(
...
+    ys = torch.linspace(0, height - 1, height, device=device, dtype=torch.float)
+    # Fix TracerWarning
     if normalized_coordinates:
-        xs = torch.linspace(-1, 1, width, device=device, dtype=torch.float)
-        ys = torch.linspace(-1, 1, height, device=device, dtype=torch.float)
-    else:
-        xs = torch.linspace(0, width - 1, width, device=device, dtype=torch.float)
-        ys = torch.linspace(0, height - 1, height, device=device, dtype=torch.float)
+        xs = (xs / (width - 1) - 0.5) * 2
+        ys = (ys / (height - 1) - 0.5) * 2
```
Comment (edgarriba) [inline kornia/utils/grid.py:37]: a better option would be to reuse[`K.normalize_pixel_coordinates`](https://kornia.readthedocs.io/en/latest/geometry.conversions.html#kornia.geometry.conversions.normalize_pixel_coordinates)

**Review by edgarriba — APPROVED** [review]
LGTM. Fix couple of comments

File: kornia/utils/grid.py (hunk 3 above)
Comment (shijianjian) [reply kornia/utils/grid.py:41]: This will not suppress the ```TracerWarning``` since it made new tensors internally. I would keep using the current implementations.

Also, I think we will need to try getting rid of new tensors in our functions.

Comment (edgarriba) [reply kornia/utils/grid.py:41]: ok, then we need to fix `K.geometry.normalize_pixel_coordinates`, right ?

**Review by edgarriba — APPROVED** [review]

File: kornia/utils/grid.py (hunk 3 above)
Comment (shijianjian) [reply kornia/utils/grid.py:41]: Theoretically, yes. The best way of addressing the JIT TracerWarnings is to initialize **NO NEW TENSORS** inside any functions, or it may lose track of the tensor trace. While I think it is not really achievable under our current library concept, since we wish to handle the python scalars for our params.

Comment (edgarriba) [reply kornia/utils/grid.py:41]: right, a bit difficult for us. However, from graph perspective I don't it will be a problem since usually new tensors are constants and for backprop I don't think it's a big deal.

**Comment by shijianjian** [comment]
@edgarriba. Please merge. The CI stuck again.
